src/SprykerEco/Zed/Braintree/Persistence/BraintreeEntityManager.php

```diff
@@ -90,7 +91,25 @@ public function updateIsShipmentOperationValue(int $idPaymentBraintree, string $
     }
 
     /**
-     * @param \Generated\Shared\Transfer\ItemTransfer[]|\ArrayObject $itemTransfers
+     * @param int $idSalesOrder
+     * @param \Generated\Shared\Transfer\BraintreePaymentTransfer $braintreePaymentTransfer
+     *
+     * @return void
+     */
+    public function updatePaymentBraintreeByIdSalesOrder(int $idSalesOrder, BraintreePaymentTransfer $braintreePaymentTransfer): void
```

Comment:
I still did not get why do we need to pass `idSalesOrder` here instead of using `BraintreePaymentTransfer.fkSalesOrder`
explanation below is not clear to me

Reply:
The reason was that `BraintreePaymentTransfer.fkSalesOrder` was not filled with an actual id. Looks like it is safe to set to `idSalesOrder`, so I removed the separate argument.